src/Compilers/CSharp/Portable/Compilation/CSharpCompilerDiagnosticAnalyzer.cs

```diff
@@ -36,7 +36,15 @@ internal override ImmutableArray<int> GetSupportedErrorCodes()
                         // We don't support configuring WRN_ALinkWarn. See comments in method "CSharpDiagnosticFilter.Filter" for more details.
                         continue;
 
+                    case (int)ErrorCode.WRN_UnreferencedField:
```

Comment:
I mean it could be done like this (sorry I don't have this cloned on this machine so it is 4 commits in the interface rather than one pushed, don't bother pulling this back into yours just do it [or don't] via your local copy): https://github.com/heejaechang/roslyn/compare/staleerror...bbarry:staleerror

@shyamnamboodiripad noted that maintaining the list is annoying; by moving it to the MessageProvider hierarchy it becomes a little simpler to manage because one would already be making changes there if they were to be adding or removing such diagnostics. I am not so sure about it though because the whole area here seems a little off. On the one side you have live diagnostics that are reported by the compiler running (as opposed to live diagnostics that are created by inspecting the trees like it would be done everywhere outside of the compiler) and at the same time you have diagnostics reported by the compiler in this clearly different set. I think it would be better if somehow the compiler diagnostic analyzer didn't deal with both of these concerns at the same time but I am not sure how to fix that (I don't grok this code well enough).